src/common.rs

```diff
@@ -149,6 +145,21 @@ pub fn solve_quadratic(c0: f64, c1: f64, c2: f64) -> ArrayVec<f64, 2> {
     result
 }
 
+/// Find real roots of linear equation.
+///
+/// Return values of x for which c0 + c1 x = 0.
+pub fn solve_linear(c0: f64, c1: f64) -> ArrayVec<[f64; 1]> {
```

Comment:
I have it like this so that it's compatible with the return types in `solve_t_for_x` and `solve_t_for_y`. It also makes it so that it follows the same format as `solve_quadratic` and `solve_cubic`. I don't have any issues with changing it to use `Option<f64>`, but I'd like your thoughts on my reasoning first.